### PR TITLE
[Issue #2495] Set points and sprint on close

### LINKED ISSUE #2495: Set points and sprint value for issues on close
### Summary

@coilysiren requested a GitHub automation that automatically sets the point value to 1 and the sprint to the current sprint if an issue is closed without those values set.

### Acceptance criteria

- [ ] If an issue is closed and sprint is none then add ticket to current sprint
- [ ] If an issue is closed and points are none then set points to 1

### PR DESCRIPTION
**Pull request by widal001**
## Summary

Adds new linter to set a default value for points and sprint if those fields are unset when an issue is closed.

Fixes #2495 

### Time to review: __5 mins__

- Adds `linters/scripts/set-points-and-sprint.sh`
- Adds associated queries:
  - `linters/queries/getFieldMetadata.graqphql`
  - `linters/queries/getItemMetadata.graphql`

This action will trigger every time an issue is closed in the repo.

### Questions

1. Do we only want to apply this to issues with a certain type (e.g. bug, task, etc.)
2. Should we post a comment to the ticket if these field values are set?
3. Are there other instances when this action shouldn't run (even if these fields are unset)?

### TODO

- [x] Update script to handle dry run mode
- [x] Add CI test to `ci-project-linters.yml`
- [x] Potentially update script based on questions above

Here's an example of this action from our sandbox repo

https://github.com/user-attachments/assets/92dc4b60-7fa6-4fc0-a621-fa468c1c01d0

Here's an example of a dry run within this repo:

Screenshot 2024-10-18 at 3 17 08 PM

### CODE REVIEW

**Comment by coilysiren** [comment]
@widal001 

> Do we only want to apply this to issues with a certain type (e.g. bug, task, etc.)

a) You can either include:

- Bug
- Task
- Enhancement
- Deliverable

b) Or exclude

- Epic

Option (a) sounds safer

> Should we post a comment to the ticket if these field values are set?

Yes that sounds great! I love clarifying comments from bots

> Are there other instances when this action shouldn't run (even if these fields are unset)?

If you can only include tick where { `author` is `users in` `this repo` with the permission level `write access` }, that would be ace. But we might get that for free? I dunno.

**Comment by widal001** [comment]
@coilysiren Thanks for the feedback above! I've implemented it in the following commits:

- [ci: Updates set-points-and-sprint.sh](https://github.com/HHS/simpler-grants-gov/pull/2519/commits/8f85c670fb48a16bc9783786ea6a44f15f8d931d)
- [ci: Adds set-points-and-sprint.sh to linter CI checks](https://github.com/HHS/simpler-grants-gov/pull/2519/commits/eefafba5794edae6cbc15faf8e177fb9da1785d5)

With two notes:

> a) You can either include:
>
> Bug
>Task
> Enhancement
> Deliverable

I'm also skipping deliverable because Deliverable is meant to be a parent to Epic, and shouldn't really ever live in a given sprint.

> If you can only include tick where { author is users in this repo with the permission level write access }, that would be ace. But we might get that for free? I dunno.

This is a bit trickier to do, but helpful to note. Maybe we can revisit if we find a _lot_ of issues being accidentally added because they were closed by open source contributors.

Here's an example of skipping an update because an issue is an Epic:

Screenshot 2024-10-21 at 11 16 27 AM

And here's [a sample comment](https://github.com/agilesix/simpler-grants-sandbox/issues/6#issuecomment-2426945182) **Note:** Unfortunately unless we switch to a different auth structure for GitHub actions, the automations tied to my PAT will continue to be shown as coming from me, hence the "beep boop".

**Comment by coilysiren** [comment]
> Unfortunately unless we switch to a different auth structure for GitHub actions, the automations tied to my PAT will continue to be shown as coming from me

@widal001 we definitely have a ticket for this... but I can't find it!

File: .github/workflows/ci-project-linters.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ jobs:
     runs-on: ubuntu-latest
     env:
       GH_TOKEN: ${{ secrets.GH_TOKEN_PROJECT_ACCESS }}
+      UNSET_ISSUE: "https://github.com/HHS/simpler-grants-gov/issues/1932"
```
Comment (coilysiren) [inline .github/workflows/ci-project-linters.yml:20]: Can you expand on what `UNSET_ISSUE` means here?

**Review by coilysiren — APPROVED** [review]
Looks good, just have a clarity comment

File: .github/workflows/ci-project-linters.yml (hunk 1 above)
Comment (widal001) [reply .github/workflows/ci-project-linters.yml:22]: Great callout! Just added some comments in this commit: [docs: Explains what UNSET_ISSUE is in ci-project-linters.yml](https://github.com/HHS/simpler-grants-gov/pull/2519/commits/7ef7e0237e089aab4b4d2233711f34ed17efd825)

**Review by coilysiren — APPROVED** [review]
Excited for this!

**Pull request merged**